GVFS/GVFS.Notifications/VFSForGit.Mac/StatusMenuItem/IPC/VFSForGitNotification.m

```diff
@@ -1,90 +1,97 @@
 #import "VFSForGitNotification.h"
 
-NSString const * IdentifierKey = @"Id";
-NSString const * EnlistmentKey = @"Enlistment";
-NSString const * EnlistmentCountKey = @"EnlistmentCount";
-NSString const * TitleKey = @"Title";
-NSString const * MessageKey = @"Message";
+NSString * const IdentifierKey = @"Id";
+NSString * const EnlistmentKey = @"Enlistment";
+NSString * const EnlistmentCountKey = @"EnlistmentCount";
+NSString * const TitleKey = @"Title";
+NSString * const MessageKey = @"Message";
 
-NSString const * AutomountTitle = @"GVFS AutoMount";
-NSString const * AutomountStartMessageFormat = @"Attempting to mount %d GVFS repos(s)";
-NSString const * AutomountSuccessMessageFormat = @"The following GVFS repo is now mounted: \n%@";
-NSString const * AutomountFailureMessageFormat = @"The following GVFS repo failed to mount: \n%@";
+NSString * const AutomountTitle = @"GVFS AutoMount";
+NSString * const AutomountStartMessageFormat = @"Attempting to mount %lu GVFS repos(s)";
+NSString * const AutomountSuccessMessageFormat = @"The following GVFS repo is now mounted: \n%@";
+NSString * const AutomountFailureMessageFormat = @"The following GVFS repo failed to mount: \n%@";
 
 @interface VFSForGitNotification()
 
-@property (readwrite, nonnull) NSString *title;
-@property (readwrite, nonnull) NSString *message;
+@property (readwrite) NSString *title;
+@property (readwrite) NSString *message;
 
 @end
 
 @implementation VFSForGitNotification
 
-+ (BOOL) tryCreateNotification:(VFSForGitNotification *_Nonnull *_Nonnull) notification
-                          from:(NSDictionary *_Nonnull) jsonMessage
-                         error:(NSString *_Nonnull *_Nonnull) error
++ (BOOL)tryValidateMessage:(NSDictionary *_Nonnull)jsonMessage
+         buildNotification:(VFSForGitNotification *_Nonnull *_Nonnull)notification
+                     error:(NSString *_Nonnull __autoreleasing *_Nonnull)error
```

Comment:
Usually an "error" parameter is of NSError type, not NSString. In objc, NSError objects are very common and preferred over throwing exceptions generally.

Reply:
Makes sense to keep it consistent, updated all errors to NSError type.